workshops/hcload/README.md

```diff
@@ -0,0 +1,465 @@
+
+Now, instead of logging the response from our library, we want to return it to the calling function. We'll log it from the calling function instead.
+
+```js
+import { connect, disconnect } from 'https://deno.land/x/ngrok@2.2.3/mod.ts'
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+import ky from 'https://unpkg.com/ky/index.js'
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+
+export default async function (filePath: string): Promise<string> { // <--- Changed
+    return new Promise(async resolve => { // <--- Changed
+        const fileBase = path.parse(filePath).base
+
+        const app = new Application()
+        app.use(async (context: any) => {
+            await send(context, filePath, { root: '/' })
+        })
+
+        app.addEventListener("listen", async ({ port }) => {
+            const ngrokUrl = `https://${await connect({ protocol: 'http', port })}/${fileBase}`
+
+            // @ts-ignore
+            let response: string[] = await ky.post('https://cdn.hackclub.com/api/new', { json: [ngrokUrl] }).json()
+
+            disconnect()
+            return resolve(response[0])
+        })
+
+        await app.listen({ port: 20685 })
+    }) as Promise<string> // <--- Changed
+}
+```
+
+Since the response is being returned inside an event listener, we'll need to wrap the returned value in a promise. 
+Make sure you
+    - Change the return type from `Promise<void>` to `Promise<string>`
+    - `return resolve(response[0])` AFTER you disconnect()
+
+In `hcload.ts`:
+
+```js
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+import hcload from "./mod.ts"
+
+console.log(await hcload(path.resolve("./LICENSE")))
+Deno.exit(0)
+```
+
+Note that we can now simply Deno.exit(0) after we get and log the URL -- we no longer have to exit the program manually using CTRL + C.
+
+EXCELLENT! We've successfully made the library, and we've also used it in `hcload.ts`. 
+
+![Hermione, Ron and Seamus cheering](https://cloud-pq5lbfiab.vercel.app/1cheer2.gif)
+
+## Part 4: Programming the CLI
+
+Now that our library is ready, let's make `hcload.ts` use arguments from the CLI instead of just hardcoding the LICENSE file. 
+
+- First, let's import [Yargs](https://yargs.js.org/). Yargs makes it easy to parse command-line arguments.
+
+```js
+import Yargs from "https://deno.land/x/yargs@v16.1.1-deno/deno.ts"
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+import hcload from "./mod.ts"
+
+console.log(await hcload(path.resolve("./LICENSE")))
+Deno.exit(0)
+```
+
+Next, let's parse the args using Yargs
+
+```js
+import Yargs from "https://deno.land/x/yargs@v16.1.1-deno/deno.ts"
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+import hcload from "./mod.ts"
+
+const args = Yargs(Deno.args)
```

Comment:
Same here re: context. I think it would actually be clearer if you said "just after all the import statements in `hcload.ts` and then just included this changed code, instead of dumping the whole file